__tests__/app.test.js

```diff
@@ -284,3 +278,110 @@ describe("POST /api/articles/article_id/comments", () => {
 
+describe("PATCH /api/articles/:article_id", () => {
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+    const articleWithId1 = {
+        title: "Living in the shadow of a great man",
+        topic: "mitch",
+        author: "butter_bridge",
+        body: "I find this existence challenging",
+        created_at: 1594329060000,
+        votes: 100,
+        article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+    }
+    test("Should return 200 and update comment", () => {
+        jest.spyOn(models, 'fetchPatchArticleByArticleId').mockResolvedValueOnce({
+            rows: [
+                {
+                    article_id: 1,
+                    title: 'Living in the shadow of a great man',
+                    topic: 'mitch',
+                    author: 'butter_bridge',
+                    body: "I find this existence challenging",
+                    votes: articleWithId1.votes + 10,
+                    article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+                },
+            ],
+        });
+        return request(app).patch("/api/articles/1").send({ incVotes: 10 }).expect(200).then((response) => {
+            const parsedArticle = JSON.parse(response.text);
+            const updatedArticle = parsedArticle.response.rows[0];
+            expect(updatedArticle.article_id).toBe(1);
+            expect(updatedArticle.title).toBe(articleWithId1.title);
+            expect(updatedArticle.topic).toBe(articleWithId1.topic);
+            expect(updatedArticle.body).toBe(articleWithId1.body);
+            expect(updatedArticle.votes).not.toBe(articleWithId1.votes);
+            expect(updatedArticle.votes).toBe(articleWithId1.votes + 10);
+            expect(updatedArticle.article_img_url).toBe(articleWithId1.article_img_url);
+        })
+    });
+    test("Should respond with 400 and send an error message if invalid id input", () => {
+        return request(app).patch("/api/articles/0").send({
+            incVotes: 10
+        }).expect(400).then((response) => {
+            const text = response.error.text;
+            expect(text).toEqual("{\"status\":400,\"error\":\"Bad Request\",\"message\":\"Invalid query\"}");
+        })
+    });
+    test("Should respond with 404 and send an error message if valid id but does not exist in database", () => {
+        return request(app).patch("/api/articles/123").send({
+            incVotes: 10
+        }).expect(404).then((response) => {
+            const text = response.error.text;
+            expect(text).toEqual("{\"status\":404,\"error\":\"Not Found\",\"message\":\"Not found\"}");
+        });
+    });
+    test("Should respond with 500 and send an error message in case of an error", () => {
+        jest.spyOn(models, 'fetchPatchArticleByArticleId').mockRejectedValue(new Error("Database query error"));
```

Comment:
This is quite interesting actually. This would be the only way to test for a 500 and can be useful.

I did say to remove mocking from your tests however in this case it is totally justified! Good job!